redfish-core/lib/processor.hpp

```diff
@@ -47,6 +47,55 @@ constexpr std::array<const char*, 2> processorInterfaces = {
     "xyz.openbmc_project.Inventory.Item.Cpu",
     "xyz.openbmc_project.Inventory.Item.Accelerator"};
 
+/**
+ * @brief Workaround to handle DCM (Dual-Chip Module) package for Redfish
+ *
+ * Make sure processor modeled as dual chip module ("dcmN-cpuN"),
+ * If yes then, replace Redfish processor id as "dcmN/cpuN" and check with
+ * given object path because Redfish does not support chip module concept.
+ *
+ * @param[in] processorId - The Redfish processor Id
+ * @param[in] objectPath  - The D-Bus object path that contain the processor
+ *                          segment
+ *
+ * @return true if matched with the given object path else false.
+ *
+ * @note Inventory modeled as "dcmN/cpuN" to support DCM so wherever using
+ *       Redfish processor id as "dcmN-cpuN" then this function (it support
+ *       both SCM and DCM) can be used for the inventory processor object
+ *       path validation.
+ */
+inline bool
+    isProcObjectMatched(const std::string& processorId,
+                        const sdbusplus::message::object_path& objectPath)
+{
+    bool isMatched = false;
+    if (processorId.find("dcm") != std::string::npos)
+    {
+        std::size_t delimiterPos = processorId.find('-');
+        if (delimiterPos != std::string::npos)
+        {
+            std::string procParent{processorId.substr(0, delimiterPos)};
+            std::string procId{
+                processorId.substr(delimiterPos + 1, processorId.length())};
```

Comment:
Don't we need to check the size of processorId ?

https://github.com/openbmc/bmcweb/blob/master/COMMON_ERRORS.md#4-walking-off-the-end-of-a-string

Also, I know this is the same but I don't see "std::string procId{" anywhere else in bmcweb..

https://github.com/openbmc/bmcweb/search?q=substr

    std::string procId = processorId.substr(delimiterPos + 1, processorId.length())

Reply:
> Not required, are you seeing any issue with that?

What if your processorId looks like "dcmN-" ? 
We have these checks not for the good case but for the bad case so bmcweb doesn't throw an exception 
Still think we should add a check 

ttps://github.com/openbmc/bmcweb/blob/master/COMMON_ERRORS.md#4-walking-off-the-end-of-a-string is this exact case

Reply:
> What if your processorId looks like "dcmN-" ?

**procId** will be empty in that case and it won't match in below `if` and bmcweb will return below errors:
```
if ((objectPath.parent_path().filename() == procParent) &&
    (objectPath.filename() == procId))
{   
    isMatched = true;
}
```

Error:
```
curl -k -H "X-Auth-Token: $bmc_token" -X GET https://${bmc}/redfish/v1/Systems/system/Processors/dcm0-
{
  "error": {
    "@Message.ExtendedInfo": [
      {
        "@odata.type": "#Message.v1_1_1.Message",
        "Message": "The requested resource of type Processor named dcm0- was not found.",
        "MessageArgs": [
          "Processor",
          "dcm0-"
        ],
        "MessageId": "Base.1.8.1.ResourceNotFound",
        "MessageSeverity": "Critical",
        "Resolution": "Provide a valid resource identifier and resubmit the request."
      }
    ],
    "code": "Base.1.8.1.ResourceNotFound",
    "message": "The requested resource of type Processor named dcm0- was not found."
  }
}
```

Reply:
Still think we need to check here, but not sure I have the energy to fight here